### DISCUSSION TITLE
Angular 18 & Zoneless

### DISCUSSION THREAD

**Discussion by rw6bsc** [gund/ng-dynamic-component#515] [General]
I'm on a project where upgrading to Angular 18 is tempting for the experimental zoneless feature. As of right now we can't upgrade to v18 and try it, because we are blocked by outdated dependencies. 
However, since our project utilizes ndc a lot, I was wondering how well this lib harmonizes with zonless.

Has anyone tried it, yet?
Anything to watch out for when migrating?

I will set up a poc project for this, once I find the time.

**Comment by gund** [comment]
This lib does not use or depend on Zone in any way so it should not matter if your app is zoneless.
But if you will encounter errors related to this lib compatibility without Zone - please feel free to report it as issue here.

**Comment by NicolaBelliardi** [comment]
Upgrading to Angular 18 with the experimental zoneless feature does sound promising, especially for performance improvements. If your project heavily relies on ndc (assuming you mean ngrx/store or a similar library), compatibility with zoneless Angular is a crucial consideration.

While the experimental zoneless mode aims to reduce overhead and improve performance by removing NgZone, it's important to note potential compatibility issues with third-party libraries like ndc. Compatibility with zoneless Angular depends on how ndc manages asynchronous operations and handles change detection outside of NgZone.

For expert guidance on navigating Angular upgrades and optimizing performance with zoneless mode, consider consulting our team of [Angular developers.](https://www.softsuave.com/hire-angularjs-developers)

**Reply by gund** [reply]
> assuming you mean ngrx/store or a similar library

Do you realize this library has nothing to do with ngrx/store?
